In [23]:
import pandas as pd 
import numpy as np 

In [24]:
df = pd.read_csv("Walmart.csv")
df

Store        Date  Weekly_Sales  Holiday_Flag  Temperature  Fuel_Price  \
0         1  05-02-2010    1643690.90             0        42.31        2.57   
1         1  12-02-2010    1641957.44             1        38.51        2.55   
2         1  19-02-2010    1611968.17             0        39.93        2.51   
3         1  26-02-2010    1409727.59             0        46.63        2.56   
4         1  05-03-2010    1554806.68             0        46.50        2.62   
...     ...         ...           ...           ...          ...         ...   
6430     45  28-09-2012     713173.95             0        64.88        4.00   
6431     45  05-10-2012     733455.07             0        64.89        3.98   
6432     45  12-10-2012     734464.36             0        54.47        4.00   
6433     45  19-10-2012     718125.53             0        56.47        3.97   
6434     45  26-10-2012     760281.43             0        58.85        3.88   

        CPI  Unemployment  
0    211.10          8.11  
1    211.24          8.11  
2    211.29          8.11  
3    211.32          8.11  
4    211.35          8.11  
...     ...           ...  
6430 192.01          8.68  
6431 192.17          8.67  
6432 192.33          8.67  
6433 192.33          8.67  
6434 192.31          8.67  

[6435 rows x 8 columns]

Data has:
- 8 columns , 6435 rows.
- Each row shows the sales of Walmart in a week. 

In [25]:
df.duplicated().sum()

0

No duplicated row

In [26]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6435 entries, 0 to 6434
Data columns (total 8 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Store         6435 non-null   int64  
 1   Date          6435 non-null   object 
 2   Weekly_Sales  6435 non-null   float64
 3   Holiday_Flag  6435 non-null   int64  
 4   Temperature   6435 non-null   float64
 5   Fuel_Price    6435 non-null   float64
 6   CPI           6435 non-null   float64
 7   Unemployment  6435 non-null   float64
dtypes: float64(5), int64(2), object(1)
memory usage: 402.3+ KB


Data description:
- Store: The store number
- Date: The week of sales in USD
- Weekly_Sales: Sales for the given storeo 
- Holiday_Flag: Whether the week is a special holiday week 1 – Holiday week 0 – Non-holiday week
- Temperature: Temperature on the day of sale in Fahrenheit
- Fuel_Price: Cost of fuel in the region
- CPI :Prevailing consumer price index
- Unemployment: Prevailing unemployment rate

In [27]:
df['Date'].apply(lambda x: type(x))
#Need to convert to datetime 


0       <class 'str'>
1       <class 'str'>
2       <class 'str'>
3       <class 'str'>
4       <class 'str'>
            ...      
6430    <class 'str'>
6431    <class 'str'>
6432    <class 'str'>
6433    <class 'str'>
6434    <class 'str'>
Name: Date, Length: 6435, dtype: object

Need to convert Date column to datetime type 

In [36]:
df['Temperature'].sample()

737   80.93
Name: Temperature, dtype: float64

Convert temperature to Celsius for convinience.

In [28]:
pd.options.display.float_format= "{:.2f}".format

In [29]:
df

Store        Date  Weekly_Sales  Holiday_Flag  Temperature  Fuel_Price  \
0         1  05-02-2010    1643690.90             0        42.31        2.57   
1         1  12-02-2010    1641957.44             1        38.51        2.55   
2         1  19-02-2010    1611968.17             0        39.93        2.51   
3         1  26-02-2010    1409727.59             0        46.63        2.56   
4         1  05-03-2010    1554806.68             0        46.50        2.62   
...     ...         ...           ...           ...          ...         ...   
6430     45  28-09-2012     713173.95             0        64.88        4.00   
6431     45  05-10-2012     733455.07             0        64.89        3.98   
6432     45  12-10-2012     734464.36             0        54.47        4.00   
6433     45  19-10-2012     718125.53             0        56.47        3.97   
6434     45  26-10-2012     760281.43             0        58.85        3.88   

        CPI  Unemployment  
0    211.10          8.11  
1    211.24          8.11  
2    211.29          8.11  
3    211.32          8.11  
4    211.35          8.11  
...     ...           ...  
6430 192.01          8.68  
6431 192.17          8.67  
6432 192.33          8.67  
6433 192.33          8.67  
6434 192.31          8.67  

[6435 rows x 8 columns]

In [30]:
df.describe()

Store  Weekly_Sales  Holiday_Flag  Temperature  Fuel_Price     CPI  \
count 6435.00       6435.00       6435.00      6435.00     6435.00 6435.00   
mean    23.00    1046964.88          0.07        60.66        3.36  171.58   
std     12.99     564366.62          0.26        18.44        0.46   39.36   
min      1.00     209986.25          0.00        -2.06        2.47  126.06   
25%     12.00     553350.10          0.00        47.46        2.93  131.74   
50%     23.00     960746.04          0.00        62.67        3.44  182.62   
75%     34.00    1420158.66          0.00        74.94        3.73  212.74   
max     45.00    3818686.45          1.00       100.14        4.47  227.23   

       Unemployment  
count       6435.00  
mean           8.00  
std            1.88  
min            3.88  
25%            6.89  
50%            7.87  
75%            8.62  
max           14.31

In [31]:
df.isnull().sum()

Store           0
Date            0
Weekly_Sales    0
Holiday_Flag    0
Temperature     0
Fuel_Price      0
CPI             0
Unemployment    0
dtype: int64

No missing values. The data is fully collected.  
- Store: this data is conducted from store 1 to store 45 
- Weekly_Sales: the average weekly sales among the stores is 960746.04 $. The prime week can be up to nearly 4 Million $.
- Holiday_Flag: barely binary value. 
- Temperature: customer could also go out for a shopping even in minimal temperature of -2.06 Fahrenheit (approx -18 Celsius degree)
- Fuel_Price, ,CPI, Unemplyment: normal ratio 

$\rightarrow$ Nothing seems abnormal. 
